nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SetSNMP.java

```diff
@@ -105,6 +113,8 @@ public void onTrigger(final ProcessContext context, final ProcessSession process
                 getLogger().error("Failed to send request to the agent. Check if the agent supports the used version.");
                 processError(context, processSession, flowFile);
             }
+        } else {
+            throw new ProcessException("No incoming flowfile found.");
```

Comment:
Not quite following the reason for throwing a ProcessException here, is it not acceptable for the processor to be triggered without input files?

Reply:
No, the processor gets the OIDs to modify as flowfile attributes. Without a flowfile we should let the user know it won't work.